File: kotlin/src/main/kotlin/advertisements/advertisement/domain/AdvertisementRepository.kt
package advertisements.advertisement.domain

import advertisements.advertisement.domain.value_object.AdvertisementId
import advertisements.advertisement.domain.value_object.ActiveAdvertisements
import advertisements.user.domain.MemberUser

interface AdvertisementRepository {
    fun save(advertisement: Advertisement)
    fun findById(id: AdvertisementId): Advertisement?
    fun activeAdvertisementsByMember(member: MemberUser): ActiveAdvertisements
    fun delete(advertisement: Advertisement)
}

File: typescript/src/advertisements/advertisement/domain/AdvertisementRepository.ts
import { Advertisement } from "./Advertisement";
import {AdvertisementId} from "./value-object/AdvertisementId";
import {MemberNumber} from "../../user/domain/value-object/MemberNumber";
import {MemberUser} from "../../user/domain/MemberUser";
import {ActiveAdvertisements} from "./value-object/ActiveAdvertisements";

export interface AdvertisementRepository {

  save(name: Advertisement): Promise<void>;

  findById(id: AdvertisementId): Promise<Advertisement | null>;

  activeAdvertisementsByMember(member: MemberUser): Promise<ActiveAdvertisements>;

  delete(advertisement: Advertisement): Promise<void>;
}
